flang/test/Lower/dense-attributed-array.f90

```diff
@@ -0,0 +1,18 @@
+! RUN: bbc --emit-fir %s -o - | FileCheck %s
+
+! Test generation of dense attributed global array. Also, make sure there are
+! no dead ssa assignments.
+module mm
+  integer, parameter :: qq(3) = [(i,i=51,53)]
+end
+subroutine ss
+  use mm
+  n = qq(3)
+end
+!CHECK: fir.global @_QMmmECqq(dense<[51, 52, 53]> : tensor<3xi32>) constant : !fir.array<3xi32>
```

Comment:
Could you add `CHECK-NOT` directives for `fir.undefined` and `fir.insert_value` ops before the `fir.global` `CHECK`?

Reply:
If there is anything extra than these lines, the test will fail right?  Isn't having a test with all !CHECK lines like that?

Reply:
This test passes with a compiler that does not have this fix.  The documentation for `CHECK-NOT` is here: https://llvm.org/docs/CommandGuide/FileCheck.html#the-check-not-directive